tests/integration/targets/mso_schema_template_contract_filter/tasks/main.yml

```diff
@@ -350,57 +350,6 @@
     - cm_add_contract_filter_both_way.current.displayName == "Contract 1"
     - cm_add_contract_filter_both_way.current.filterType == "both-way"
 
-- name: Change Contract type both_way Filter type consumer-to-provider(normal_mode)
-  mso_schema_template_contract_filter:
-    <<: *contract_present
-    contract: Contract 1
-    contract_filter_type: both-way
-    filter: Filter 4
-    filter_type: consumer-to-provider
-    state: present
-  ignore_errors: yes
-  register: nm_both_way_and_consumer_to_provider
-
-- name: Verify nm_both_way_and_consumer_to_provider
-  assert:
-    that:
-    - nm_both_way_and_consumer_to_provider is not changed
-    - nm_both_way_and_consumer_to_provider.msg == "You are adding 'one-way' filters to a 'both-way' contract"
-
-- name: Change Contract type both_way Filter type provider-to-consumer(normal_mode)
-  mso_schema_template_contract_filter:
-    <<: *contract_present
-    contract: Contract 1
-    contract_filter_type: both-way
-    filter: Filter 4
-    filter_type: provider-to-consumer
-    state: present
-  ignore_errors: yes
-  register: nm_both_way_and_provider_to_consumer
-
-- name: Verify nm_both_way_and_provider_to_consumer
-  assert:
-    that:
-    - nm_both_way_and_provider_to_consumer is not changed
-    - nm_both_way_and_provider_to_consumer.msg == "You are adding 'one-way' filters to a 'both-way' contract"
-
-- name: Change Contract type one_way Filter type both way(normal_mode)
-  mso_schema_template_contract_filter:
-    <<: *contract_present
-    contract: Contract 1
-    contract_filter_type: one-way
-    filter: Filter 4
-    filter_type: both-way
-    state: present
-  ignore_errors: yes
-  register: nm_one_way_and_both_way
-
-- name: Verify nm_one_way_and_both_way
-  assert:
-    that:
-    - nm_one_way_and_both_way is not changed
-    - nm_one_way_and_both_way.msg == "You are adding 'both-way' filters to a 'one-way' contract"
-
```

Comment:
Why are we removing those tests? If it is not failing for recent version just add a version condition.

Reply:
Because the error handling is removed from code when the deducing of contract_filter_type was introduced.

The check looked at the input value of contract_filter_type (now deprecated), which could deviate from the actual configuration. The error message that the MSO/NDO API returns is clear enough when someone tries to configure this change in a version prior to 3.3. Later versions support changing this type.